### DISCUSSION TITLE
Archive failing on CodeMagic while working locally

### DISCUSSION THREAD

**Discussion by galvin59** [codemagic-ci-cd/codemagic-docs#1344] [Q&A] [answered]
Hi,

I have a flutter project which compiles fine (iOS version) on both CodeMagic and my local machine

However, as soon as I activate iOS Code signing on code magic, the build fails with the following (minimal ...) error : 
Running pod install...                                             44.2s
Running Xcode build...                                          
Xcode archive done.                                         100.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** ARCHIVE FAILED **
Xcode's output:
↳
    Writing result bundle at path: /var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.iSUr6V/flutter_ios_build_temp_dirmhafN2/temporary_xcresult_bundle
    error: the following command failed with exit code 1 but produced no further output
    Run custom shell script 'Run Script'

I have no other information in the build logs, apart from other warnings.
If I create an archive on Xcode on my local machine, with the same certificate, it works smoothly.

Any idea of what is happening ? Or any clue on how to get more details about this error ?
Thanks

Julien

**Comment by himesh-cm** [comment]
Hi @galvin59, Please share the build id, so we can look into the issue further.

**Comment by galvin59** [comment]
Thanks a lot
here it is : 63d0ff2d9b2bc5f373b0b1d3

**Comment by ramirezblindside** [comment]
Wanted to stop in and confirm that I am having this issue too since ~16hrs ago.

Edit: Here is a build ID of mine that is failing - 63d22f481f9fe032e8f3cc7b

**Reply by ramirezblindside** [reply]
The same code, same flutter version and same xcode version on my local machine can successfully archive. It seems that it only fails on codemagic for me.

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/87979934/214784534-f3807288-ccfd-4e4b-9e89-3121bc9c321a.png">

**Reply by ramirezblindside** [reply]
I just noticed that locally I am running with the Debug flag and on codemagic we are building with Release flag

**Reply by ramirezblindside** [reply]
@himesh-cm any ideas on what to look at? Unfortunately my build pipeline is completely blocked by this issue.

**Reply by himesh-cm** [reply]
Checking

**Reply by himesh-cm** [reply]
Hi @ramirezblindside, i see a bunch of warnings related to `SnowplowTracker` and `flutter_inappwebview` but I am not sure if they are the root cause for the failure.

Can you please run the last successful commit id `42995e0` again for debugging purposes?

Also, I noticed the last successful build was on `Xcode 14.0`, so if you can run the latest commit on the same Xcode version and share the build id so that I can look into it further?

**Reply by ramirezblindside** [reply]
@himesh-cm I ended up resolving this by upgrading my codebase to Flutter 3.7.0.
The warnings you mentioned were just warnings.
Flutter 3.7.0 and XCode 14.0 works for me. No other changes.

**Comment by galvin59** [comment]
I tried to go for Flutter 3.7.0 and Xcode 14 but I had no luck :(
Still fails and this time with even fewer information : 
Running Xcode build...                                          
Xcode archive done.                                         84.7s
Failed to build iOS app
Uncategorized (Xcode): Exited with status code 1
Encountered error while archiving for device.
Build failed :|
Failed to build for iOS

@himesh-cm  did you have a chance to look at my build logs ? For me too, the integration pipe is blocked by this problem :(
Thanks a lot !

**Reply by himesh-cm** [reply]
Hi @galvin59,
Was there a recent change in the codebase? A quick check would be to run the successful commit id `4d804e1` again to see if the build fails.

Errors
```
/Users/builder/clone/monbergerlocal/ios/Pods/TOCropViewController/Objective-C/TOCropViewController/TOCropViewController.m:87:14: warning: 'automaticallyAdjustsScrollViewInsets' is deprecated: first deprecated in iOS 11.0 - Use UIScrollView's contentInsetAdjustmentBehavior instead [-Wdeprecated-declarations]
            self.automaticallyAdjustsScrollViewInsets = NO;
                 ^
    In module 'UIKit' imported from /Users/builder/clone/monbergerlocal/ios/Pods/Target Support Files/TOCropViewController/TOCropViewController-prefix.pch:2:
    /Applications/Xcode-14.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:260:34: note: 'automaticallyAdjustsScrollViewInsets' has been explicitly marked deprecated here
    @property(nonatomic,assign) BOOL automaticallyAdjustsScrollViewInsets API_DEPRECATED("Use UIScrollView's contentInsetAdjustmentBehavior instead", ios(7.0,11.0),tvos(7.0,11.0)); // Defaults to YES
                                     ^
    /Users/builder/clone/monbergerlocal/ios/Pods/TOCropViewController/Objective-C/TOCropViewController/TOCropViewController.m:1299:36: warning: 'topLayoutGuide' is deprecated: first deprecated in iOS 11.0 - Use view.safeAreaLayoutGuide.topAnchor instead of topLayoutGuide.bottomAnchor [-Wdeprecated-declarations]
                statusBarHeight = self.topLayoutGuide.length;
                                       ^
    In module 'UIKit' imported from /Users/builder/clone/monbergerlocal/ios/Pods/Target Support Files/TOCropViewController/TOCropViewController-prefix.pch:2:
    /Applications/Xcode-14.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:505:58: note: 'topLayoutGuide' has been explicitly marked deprecated here
    @property(nonatomic,readonly,strong) id<UILayoutSupport> topLayoutGuide API_DEPRECATED("Use view.safeAreaLayoutGuide.topAnchor instead of topLayoutGuide.bottomAnchor", ios(7.0,11.0), tvos(7.0,11.0));
```

**Reply by himesh-cm** [reply]
Configuration-wise, everything looks fine.
Apologies i missed your message earlier and not getting back to you sooner.

**Reply by himesh-cm** [reply]
There are some additional errors with regards to `The app icon set "AppIcon" has 6 unassigned children` you should see it under the `Building iOS` step.

**Reply by galvin59** [reply]
No pb and thanks  lot for talking the time to investigate the issue
I just did a new build without code signing (disabled in settings)
This build runs just fine (63d2a3740fa34feb19b82837)
I then tried with the exact same settings but with code signing enabled. It fails (63d2a4b9abc34e90b7380e07) and what is even more surprising is that I get even less error message this time (see log below) ... so I cannot see any of the errors you mentioned previsously
Any idea ?

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/26-01-23_bocc4q5z.keychain-db
Searching for files matching /Users/builder/clone/xxx/ios/**/*.xcodeproj
Completed configuring code signing settings
 - Using profile "XC solutions spike mbl ios_app_store XXX" [6b7bb020-fe88-42fa-8af8-f0b9055ae7df] for target "Runner" [Debug] from project "Runner"
 - Using profile "XC solutions spike mbl ios_app_store XXX" [6b7bb020-fe88-42fa-8af8-f0b9055ae7df] for target "Runner" [Profile] from project "Runner"
 - Using profile "XC solutions spike mbl ios_app_store XXX" [6b7bb020-fe88-42fa-8af8-f0b9055ae7df] for target "Runner" [Release] from project "Runner"
Generated options for exporting the project
 - Method: app-store
 - Provisioning Profiles:
     - xxxx: XC solutions xxx xxx xxx
 - Signing Certificate: Apple Distribution
 - Signing Style: manual
 - Team Id: XXXX
Saved export options to /Users/builder/export_options.plist

> flutter build ipa --release --export-options-plist /Users/builder/export_options.plist --build-name=1.0.20 --build-number=20
Running "flutter pub get" in monbergerlocal...                   1,439ms

💪 Building with sound null safety 💪

Archiving solutions.xxx.mbl...
Automatically signing iOS for device deployment using specified development team in Xcode project: PTP4GC5XG5
Running pod install...

**Reply by galvin59** [reply]
apparently there was a slowdown with logs, I took several minutes to show up on the site. Now I have got them.
From what I see the errors you mentioned are only warnings (app icons, deprecated stuffs)
Moreover, the fact that the project builds just fine without code signing and fails as soon as code signing is activated tends to show the problem is not on the code side, right ?

**Comment by galvin59** [comment, marked as answer]
Problem fixed finally. I started a new project from scratch and realized that the problem was coming from using different flavor and passing them in the command line

**Reply by himesh-cm** [reply]
Hi @galvin59, Thanks for sharing the solution.
